tests/test_psf.py

```diff
@@ -360,6 +365,27 @@ def testValidateGalsimInterpolant(self):
             self.assertTrue(_validateGalsimInterpolant(f"galsim.{interp}"))
             self.assertTrue(eval(f"galsim.{interp}"))
 
+    def testKernelSize(self):  # TODO: Remove this test in DM-36311.
+        warnings.filterwarnings("ignore", category=FutureWarning)
```

Comment:
I think you can do with with a `with self.assertWarns:` block, which will both check that the warning appears and make sure the warnings filter is disabled properly at the end.